eva-tools/src/main/java/embl/ebi/variation/eva/vcfdump/BiodataVariantToVariantContextConverter.java

```diff
@@ -0,0 +1,200 @@
+                        + regionStart + "-" + regionEnd, e);
+            }
+        }
+        String nucleotide = getNucleotideFromRegionSequence(contextNucleotidePosition, regionStart, regionSequence);
+        return nucleotide;
+    }
+
+    private String getNucleotideFromRegionSequence(int nucleotidePosition, int regionStart, String regionSequence) {
+        int relativePosition = nucleotidePosition - regionStart;
+        return regionSequence.substring(relativePosition, relativePosition + 1);
+    }
+
+    private Set<Genotype> getGenotypes(Variant variant, String[] allelesArray) {
+        Set<Genotype> genotypes = new HashSet<>();
+
+        Allele[] variantAlleles =
+                {Allele.create(allelesArray[0], true), Allele.create(allelesArray[1]), Allele.create(Allele.NO_CALL, false)};
+
+        for (VariantSource source : sources) {
+            List<VariantSourceEntry> variantStudyEntries =
+                    variant.getSourceEntries().values().stream().filter(s -> s.getStudyId().equals(source.getStudyId())).collect(Collectors.toList());
+            for (VariantSourceEntry variantStudyEntry : variantStudyEntries) {
+                genotypes = getStudyGenotypes(genotypes, variantAlleles, variantStudyEntry);
+            }
+        }
+        return genotypes;
+    }
+
+    private Set<Genotype> getStudyGenotypes(Set<Genotype> genotypes, Allele[] variantAlleles, VariantSourceEntry variantStudyEntry) {
+        for (Map.Entry<String, Map<String, String>> sampleEntry : variantStudyEntry.getSamplesData().entrySet()) {
+            String sampleGenotypeString = sampleEntry.getValue().get(GENOTYPE_KEY);
+            Genotype sampleGenotype =
+                    parseSampleGenotype(variantAlleles, variantStudyEntry.getFileId(), sampleEntry.getKey(), sampleGenotypeString);
+            genotypes.add(sampleGenotype);
+        }
+        return genotypes;
+    }
+
+    private Genotype parseSampleGenotype(Allele[] variantAlleles, String fileId, String sampleName, String sampleGenotypeString) {
+        // use opencb biodata-models Genotype class for parsing the genotype string and get the list of genotype allele indexes
+        org.opencb.biodata.models.feature.Genotype genotype =
+                new org.opencb.biodata.models.feature.Genotype(sampleGenotypeString, variantAlleles[0].getBaseString(), variantAlleles[1].getBaseString());
+        List<Allele> genotypeAlleles = new ArrayList<>(2);
+        for (int index : genotype.getAllelesIdx()) {
+            // every allele not 0 or 1 will be considered no call
+            if (index == -1 || index > NO_CALL_ALLELE_INDEX) {
+                index = NO_CALL_ALLELE_INDEX;
+            }
+            genotypeAlleles.add(variantAlleles[index]);
+        }
+
+        GenotypeBuilder builder = new GenotypeBuilder()
+                .name(getFixedSampleName(fileId, sampleName))
+                .alleles(genotypeAlleles);
```

Comment:
I think the only remaining issue in this PR is to set the phase again using `.phased(genotype.isPhased())`